pkg/apis/hive/v1alpha1/clusterdeployment_types.go

```diff
@@ -80,6 +80,10 @@ type ClusterDeploymentSpec struct {
 
 	// PreserveOnDelete allows the user to disconnect a cluster from Hive without deprovisioning it
 	PreserveOnDelete bool `json:"preserveOnDelete,omitempty"`
+
+	// CreateCertificates, if true, causes the cluster deployment to request TLS serving certificates
+	// for the api and router shards on the target cluster.
+	CreateCertificates bool `json:"generateCertificates,omitempty"`
```

Comment:
Name mismatch there, create vs generate in the json.

Reply:
What would happen if someone wanted to pass in their own pre-existing certs? I kinda think the API should be able to accomodate that even if it's not needed right now. A system where they give us some, if they don't we generate, would be nice. This flag here would give a third option which is "I didn't pass in certs and I don't want you to generate any" but I'm not sure anyone would want that.

Reply:
@dgoodwin so i suppose you'd have a spec.controlPlaneCertBundleSecret and then the ability to bring your own CertBundle for each defined ingress?

Reply:
@dgoodwin would we expect them to give us separate certs for api/apps/[other router shards]? If so, I would assume we want something like:
```
servingCerts:
  api:
    name: blah
  ingress:
  - name: default
    secret:
      name: blah
  - name: other-ingress
    secret:
      name: blahblah
```

Reply:
could it work like this:
if api.name is a non-nil secret reference and the secret exists, then that's the 'bring your own certificates' mode
if api.name is a non-nil secret reference and the secret doesn't exist, then we're in 'lets encrypt' mode
if api.name is nil, then it's default installer behavior of self-signed certs

Reply:
Hmm, I would prefer that they explicitly tell us what they want. If you want your certs generated, you shouldn’t have to fill in anything, just say that you want them created and we’ll handle it. We may want to fill out a section in status saying which certs were generated.

Reply:
after thinking it though some more, that setup i wrote above would be a pain to reconcile with the possibility that someone could come in and edit the clusterdeployment from the lets-encrypt mode to bring-your-own mode. and the behavior would be a bit too magical for my taste...so nevermind

Reply:
i do like organizing the control plane and the ingress list together since they all deal with the external-facing pieces of the cluster. need to check with the ingress team to see if they have plans to own/lifecycle the control plane parts of the cluster as well